packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-answer.tsx

```diff
@@ -15,14 +15,38 @@ export class AtomicSmartSnippetAnswer {
   @Prop({reflect: true}) htmlContent!: string;
   @Prop({reflect: true}) innerStyle?: string;
 
-  @Event({
-    eventName: 'atomic/smartSnippet/answerRendered',
-  })
-  private answerRendered!: EventEmitter<{height: number}>;
+  @Event({bubbles: false})
+  private answerSizeUpdated!: EventEmitter<{height: number}>;
   private wrapperElement?: HTMLElement;
+  private isRendering = true;
+  private resizeObserver: ResizeObserver | undefined;
+
+  public componentWillRender() {
+    this.isRendering = true;
+  }
 
   public componentDidRender() {
-    this.answerRendered.emit({height: this.wrapperElement!.scrollHeight});
+    this.isRendering = false;
+    this.emitCurrentHeight();
+  }
+
+  public connectedCallback() {
+    if (this.wrapperElement && this.resizeObserver) {
+      this.resizeObserver.observe(this.wrapperElement);
+    }
+  }
+
+  public disconnectedCallback() {
+    this.resizeObserver?.disconnect();
```

Comment:
I saw a mention on stackoverflow of somebody getting a memory leak due to forgetting to call `disconnect`.

Reply:
yes it's important